src/Analyzers/CSharp/Analyzers/MakeStructMemberReadOnly/CSharpMakeStructMemberReadOnlyAnalyzer.cs

```diff
@@ -89,10 +109,38 @@ protected override void InitializeWorker(AnalysisContext context)
             if (blockOperation is IBlockOperation { Operations: [IThrowOperation or IExpressionStatementOperation { Operation: IThrowOperation }] })
                 return;
 
-            if (BlockOperationPotentiallyMutatesThis(owningMethod, blockOperation, cancellationToken))
+            if (BlockOperationPotentiallyMutatesThis((IMethodSymbol)context.OwningSymbol, blockOperation, cancellationToken))
```

Comment:
i'd pull this up, and do the IMethod check.  then do the other check on that imethod.